Source/WebCore/dom/WindowEventLoop.cpp

```diff
@@ -116,6 +116,11 @@ MicrotaskQueue& WindowEventLoop::microtaskQueue()
 }
 
 void WindowEventLoop::didReachTimeToRun()
+{
+    runTasks();
+}
+
+void WindowEventLoop::runTasks()
 {
     Ref protectedThis { *this }; // Executing tasks may remove the last reference to this WindowEventLoop.
```

Comment:
This should remain in WindowEventLoop::didReachTimeToRun. The caller of runTasks should be keeping WindowEventLoop object alive.